### PR TITLE
Enable encrypted logging and add tracking

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR re-enables encrypted logging and adds tracking. 738a93400395cf0eecf46a7061b189d275334a15 is the revert commit, so it may be skipped during review, so the only code to review is 8a976ae34761dbaf6131cec3bcbe9acd8644f8e6.

**To test:**
* Crash the app using instructions from #12707
* Re-open the app
* Find the crash event in Sentry from https://sentry.io/share/issue/04a76a1c7b6240c699fe4b907668716c/ and grab its uuid
* Use the uuid to verify the logs are successfully uploaded to MC

**PR submission checklist:**

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/util/EncryptedLogging.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    fun onEncryptedLogUploaded(event: OnEncryptedLogUploaded) {
+        when (event) {
+            is EncryptedLogUploadedSuccessfully -> {
+                AppLog.i(T.MAIN, "Encrypted log with uuid: ${event.uuid} uploaded successfully!")
+                analyticsTrackerWrapper.track(Stat.ENCRYPTED_LOGGING_UPLOAD_SUCCESSFUL)
+            }
+            is EncryptedLogFailedToUpload -> {
+                AppLog.e(T.MAIN, "Encrypted log with uuid: ${event.uuid} failed to upload with error: ${event.error}")
+                // Only track final errors
+                if (event.willRetry) {
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/util/EncryptedLogging.kt]: Just to double check -> shouldn't this be `!event.willRetry`?

If it's correct the way it is now, I feel like the comment might be a bit confusing.

**Review by malinajirka — COMMENTED** [review]
Thanks @oguzkocer! I'm going to test the changes now. Just dropping this comment to speed up the process in case it's actually a bug.

File: WordPress/src/main/java/org/wordpress/android/util/EncryptedLogging.kt (hunk 1 above)
Comment (oguzkocer) [reply WordPress/src/main/java/org/wordpress/android/util/EncryptedLogging.kt]: Thanks for catching that @malinajirka! Fixed in 574bbaea5e7399830c3b1600dc5e6822a9d7118a.

**Review by malinajirka — APPROVED** [review]
Thanks @oguzkocer! Works as expected ;)!

**Pull request merged**